pkg/autohealer/healer.go

```diff
@@ -198,23 +203,13 @@ func (h *Healer) run(ctx context.Context, e *common.Etcd, wg *sync.WaitGroup, nc
 						chans[instance] = &ci
 						go func(ci chan struct{}, instance string) {
 							var compute string
-							key := common.Path(h.CloudID, instance)
-						wait:
-							//	wait for correct compute-instance pair
 							for {
-								select {
-								case <-h.done:
-									return
-								case <-ci:
-									return
-								case c := <-nc:
-									if com, ok := c[key]; ok && com != "" {
-										compute = com
-										break wait
-									}
-								default:
-									nc <- map[string]string{"instance": key}
+								if com, ok := h.nodes.Get(instance); ok {
+									compute = com.(string)
+									break
 								}
+								// A trick to create a busy loop
+								time.Sleep(200 * time.Millisecond)
```

Comment:
no no no and no!

Reply:
- OK,  let's change to the idiomatic way - time.Ticker. Anw please leave a explanation rather than just leaving a statement like this.

- Why adding a sleep?

This gorountine is calculating the time difference with as many free clock cycles as the CPU has for it, which is why it uses so much CPU resources. Adding the sleep reduces the load because the gorountine rests.

This is the reason why your first healer module causes high CPU usage - 2 magic complex busy loops worth nothing.

https://github.com/vCloud-DFTBA/faythe/blob/master/pkg/autohealer/healer.go#L204

Reply:
I think just put a fixed time.sleep in code is not an acceptable way, it doesn't requires any `statements` for that.

Reply:
Give me link to prove it, at least it can reduce CPU usage :ok_hand: